sbe-tool/src/test/java/uk/co/real_logic/sbe/FieldAccessOrderCheckTest.java

```diff
@@ -0,0 +1,2953 @@
+        assertThat(bs.next().c(), equalTo(3));
+        assertThat(bs.next().c(), equalTo(4));
+        assertThat(decoder.d(), equalTo("def"));
+    }
+
+    @Test
+    void allowsEncodingAndDecodingEmptyGroupAndVariableLengthFieldsInSchemaDefinedOrder()
+    {
+        final GroupAndVarLengthEncoder encoder = new GroupAndVarLengthEncoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderEncoder);
+        encoder.a(42);
+        encoder.bCount(0);
+        encoder.d("abc");
+
+        final GroupAndVarLengthDecoder decoder = new GroupAndVarLengthDecoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderDecoder);
+        assertThat(decoder.a(), equalTo(42));
+        final GroupAndVarLengthDecoder.BDecoder bs = decoder.b();
+        assertThat(bs.count(), equalTo(0));
+        assertThat(decoder.d(), equalTo("abc"));
+        assertThat(decoder.toString(), containsString("a=42|b=[]|d='abc'"));
+    }
+
+    @Test
+    @Disabled("Our access checks are too strict to allow the behaviour in this test.")
+    void allowsReEncodingPrimitiveFieldInGroupElementAfterTopLevelVariableLengthField()
+    {
+        final GroupAndVarLengthEncoder encoder = new GroupAndVarLengthEncoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderEncoder);
+        encoder.a(42);
+        final GroupAndVarLengthEncoder.BEncoder bEncoder = encoder.bCount(2);
+        bEncoder
+            .next()
+            .c(1)
+            .next()
+            .c(2);
+        encoder.d("abc");
+        bEncoder.c(3);
+
+        final GroupAndVarLengthDecoder decoder = new GroupAndVarLengthDecoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderDecoder);
+        assertThat(decoder.a(), equalTo(42));
+        final GroupAndVarLengthDecoder.BDecoder bs = decoder.b();
+        assertThat(bs.count(), equalTo(2));
+        assertThat(bs.next().c(), equalTo(1));
+        assertThat(bs.next().c(), equalTo(3));
+        assertThat(decoder.d(), equalTo("abc"));
+    }
+
+    @Test
+    @Disabled("Our access checks are too strict to allow the behaviour in this test.")
+    void allowsReWrappingGroupDecoderAfterAccessingLength()
+    {
+        final GroupAndVarLengthEncoder encoder = new GroupAndVarLengthEncoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderEncoder);
+        encoder.a(42);
+        final GroupAndVarLengthEncoder.BEncoder bEncoder = encoder.bCount(2);
+        bEncoder
+            .next()
+            .c(1)
+            .next()
+            .c(2);
+        encoder.d("abc");
+
+        final GroupAndVarLengthDecoder decoder = new GroupAndVarLengthDecoder()
+            .wrapAndApplyHeader(buffer, OFFSET, messageHeaderDecoder);
+        assertThat(decoder.a(), equalTo(42));
+        assertThat(decoder.b().count(), equalTo(2));
+        final GroupAndVarLengthDecoder.BDecoder b = decoder.b();
+        assertThat(b.next().c(), equalTo(1));
+        assertThat(b.next().c(), equalTo(3));
```

Comment:
This should be `equalTo(2)`.

Reply:
Good spot. I'll fix that. I was very worried that the tests weren't running properly until I saw the `@Disabled` annotation. :-)

Reply:
It is fixed in 6794fc3.